app/src/main/java/com/bogdan/codeforceswatcher/features/users/UsersFragment.kt

```diff
@@ -1,51 +1,80 @@
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
 import androidx.fragment.app.Fragment
-import androidx.recyclerview.widget.LinearLayoutManager
-import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
 import com.bogdan.codeforceswatcher.R
-import com.bogdan.codeforceswatcher.epoxy.BaseEpoxyController
+import com.bogdan.codeforceswatcher.components.compose.theme.AlgoismeTheme
+import com.bogdan.codeforceswatcher.features.auth.SignInComposeActivity
+import com.bogdan.codeforceswatcher.features.auth.VerificationComposeActivity
+import com.bogdan.codeforceswatcher.features.users.compose.ProfileItemView
+import com.bogdan.codeforceswatcher.features.users.compose.UserItemView
+import com.google.accompanist.swiperefresh.SwipeRefresh
+import com.google.accompanist.swiperefresh.rememberSwipeRefreshState
 import io.xorum.codeforceswatcher.features.auth.models.UserAccount
 import io.xorum.codeforceswatcher.features.auth.redux.AuthState
 import io.xorum.codeforceswatcher.features.users.models.User
 import io.xorum.codeforceswatcher.features.users.redux.FetchUserDataType
 import io.xorum.codeforceswatcher.features.users.redux.UsersActions
 import io.xorum.codeforceswatcher.features.users.redux.UsersRequests
 import io.xorum.codeforceswatcher.features.users.redux.UsersState
-import io.xorum.codeforceswatcher.features.users.redux.UsersState.SortType.Companion.getSortType
 import io.xorum.codeforceswatcher.redux.analyticsController
 import io.xorum.codeforceswatcher.redux.states.AppState
 import io.xorum.codeforceswatcher.redux.store
 import io.xorum.codeforceswatcher.util.AnalyticsEvents
-import kotlinx.android.synthetic.main.fragment_users.*
 import tw.geothings.rekotlin.StoreSubscriber
 
-class UsersFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener, StoreSubscriber<AppState> {
+class UsersFragment : Fragment(), StoreSubscriber<AppState> {
 
-    private lateinit var spSort: AppCompatSpinner
+    private val appState: MutableState<AppState?> = mutableStateOf(null)
 
-    private val epoxyController by lazy { EpoxyController() }
-
-    override fun onRefresh() {
-        store.dispatch(
-            UsersRequests.FetchUserData(
-                FetchUserDataType.REFRESH,
-                isInitiatedByUser = true
-            )
-        )
-        analyticsController.logEvent(AnalyticsEvents.USERS_REFRESH)
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ) = ComposeView(requireContext()).apply {
+        setContent {
+            AlgoismeTheme {
+                ContentView(
+                    appState = appState,
+                    onRefresh = { onRefresh() },
```

Comment:
You can simplify this a bit with `onRefresh = ::onRefresh`. `::` allows to reference functions directly.